Tests/MongoSwiftTests/SpecTestRunner/ChangeStreamSpecTest.swift

```diff
@@ -0,0 +1,136 @@
+    }
+}
+
+internal struct TestResult: Decodable {
+    /// Describes an error received during the test
+    let error: ChangeStreamTestError?
+
+    /// An Extended JSON array of documents expected to be received from the changeStream
+    let success: [ChangeStreamTestEventDocument]?
+}
+
+internal struct ChangeStreamTestError: Decodable {
+    let code: Int
+
+    let errorLabels: [String]?
+}
+
+internal struct ChangeStreamTestEventDocument: Codable, Equatable {
+    let operationType: String
+
+    let ns: MongoNamespace?
+
+    let fullDocument: Document?
+
+    public static func == (lhs: ChangeStreamTestEventDocument, rhs: ChangeStreamTestEventDocument) -> Bool {
+        let lhsFullDoc = lhs.fullDocument?.filter { elem in
+            let key = elem.key
+            return key != "_id"
+        }
+
+        let rhsFullDoc = rhs.fullDocument?.filter { elem in
+            let key = elem.key
+            return key != "_id"
+        }
+
+        return lhsFullDoc == rhsFullDoc && lhs.ns == rhs.ns && lhs.operationType == rhs.operationType
+    }
+}
+
+internal protocol ChangeStreamSpecTest: Decodable {
+    var description: String { get }
+    var operations: [ChangeStreamAnyTestOperation] { get }
+    var result: TestResult { get }
+
+    func run(client: MongoClient,
+             seenError: Error?,
+             changeStream: ChangeStream<ChangeStreamTestEventDocument>?) throws
+}
+
+extension ChangeStreamSpecTest {
+    func run(client: MongoClient,
+             seenError: Error?,
+             changeStream: ChangeStream<ChangeStreamTestEventDocument>?) throws {
+        var seenError = seenError
+        for operation in self.operations {
+            guard let dbName = operation.database else {
+                print("The database name for running the operation is missing.")
+                return
+            }
+
+            guard let collName = operation.collection else {
+                print("The collection name for running the operation is missing.")
+                return
+            }
+
+            let db = client.db(dbName)
+            let coll = db.collection(collName)
+
+            do {
+                try operation.anyTestOperation.op.execute(client: client,
+                                                          database: db,
+                                                          collection: coll,
+                                                          session: nil)
+            } catch {
+                seenError = error
+            }
+        }
+
+        // If there was no error and result.error is set, iterate changeStream once and capture any error.
+        if seenError == nil && result.error != nil {
+            do {
+                try changeStream?.nextOrError()
+            } catch {
+                seenError = error
+            }
+        }
+
+        if seenError != nil {
+            // assert errors match expected errors
+            assertError(seenError: seenError)
```

Comment:
this conditional can be made into an `if let` and the signature of `assertError` updated to take in a non-optional.
for the success branch, you can just pass in `changeStream!` since you know it will be non-nil if the error is non-nil.